auslib/util/swagger.py

```diff
@@ -0,0 +1,119 @@
+import yaml
+import auslib
+
+from os import path
+
+AUS_DIR = path.dirname(auslib.__file__)
+
+
+class SpecBuilder():
+    def __init__(self,
+                 app,
+                 endpoints_key="paths",
+                 definitions_key="definitions",
+                 parameters_key="parameters",
+                 version_file="version.txt"):
+        self._app = app
+        self.endpoints_key = endpoints_key
+        self.definitions_key = definitions_key
+        self.parameters_key = parameters_key
+        self._main_spec = {}
+        self._spec_parts = []
+        self._version_file = version_file
+
+    def _get_version(self):
+        version = ''
+        root_path = path.dirname(AUS_DIR)
+        version_file = path.join(root_path, self._version_file)
+        if path.exists(version_file):
+            with open(version_file, 'r') as f:
+                version = f.read()
+        return version
+
+    def _get_dict(self, spec, key):
+        for k, v in spec.items():
+            if k == key:
+                return v
+            elif isinstance(v, dict):
+                self._get_dict(v, key)
+        return {}
+
+    def _load_spec(self, file_name):
+        with file(file_name, 'r') as spec_file:
+            return yaml.load(spec_file)
+
+    def _try_change_version(self, info_dict):
+        version_key = 'version'
+        if info_dict and version_key in info_dict:
+            version = self._get_version()
+            if version:
+                info_dict[version_key] = version
+
+    def add_main_spec(self, file_name):
+        """
+        Adds the current application specification file.
+
```

Comment:
If I remember correctly, the "main" spec is required to have a bunch of top level keys that won't be present in the spec "parts". If that's the case, please enhance this docstring to reflect that.